src/geometry.F90

```diff
@@ -1326,81 +1326,66 @@ recursive subroutine count_instance(univ)
 
     type(Universe), intent(in) :: univ  ! universe to search through
 
-    integer :: i                          ! index over cells
-    integer :: j, k, m                    ! indices in lattice
-    integer :: n                          ! number of cells to search
-    integer :: cell_index                 ! index in cells array
-    type(Cell),     pointer :: c          ! pointer to current cell
-    type(Universe), pointer :: next_univ  ! next universe to loop through
-    class(Lattice), pointer :: lat        ! pointer to current lattice
+    integer :: i        ! index over cells
+    integer :: j, k, m  ! indices in lattice
+    integer :: n        ! number of cells to search
 
-    n = univ % n_cells
+    n = size(univ % cells)
 
     do i = 1, n
 
-      cell_index = univ % cells(i)
-
       ! get pointer to cell
```

Comment:
This comment no longer appears to be as relevant, remove or perhaps reword as "increment cell count"?